pkg/api/validation/validation.go

```diff
@@ -649,3 +649,29 @@ func ValidateResourceName(str string) errs.ValidationErrorList {
 
 	return errs.ValidationErrorList{}
 }
+
+// ValidateLimitRange tests if required fields in the LimitRange are set.
+func ValidateLimitRange(limitRange *api.LimitRange) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if len(limitRange.Name) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("name", limitRange.Name))
+	} else if !util.IsDNSSubdomain(limitRange.Name) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("name", limitRange.Name, ""))
+	}
+	if len(limitRange.Namespace) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("namespace", limitRange.Namespace))
+	} else if !util.IsDNSSubdomain(limitRange.Namespace) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", limitRange.Namespace, ""))
+	}
+	// ensure resource names are properly qualified per docs/resources.md
+	for i := range limitRange.Spec.Limits {
+		limit := limitRange.Spec.Limits[i]
+		for k := range limit.Max {
+			allErrs = append(allErrs, ValidateResourceName(string(k))...)
+		}
+		for k := range limit.Min {
+			allErrs = append(allErrs, ValidateResourceName(string(k))...)
+		}
```

Comment:
For validation, no.

I want to be able to do the following downstream:

```
{
  "id": "limits",
  "kind": "LimitRange",
  "apiVersion": "v1beta1",
  "spec": {
    "limits": [
    {
      "type": "Pod",
      "max": {
        "memory": "1073741824",
        "cpu": "2",
      },
      "min": {
        "memory": "1048576",
        "cpu": "0.25"
      }
    },
    {
      "type": "openshift.com/ShinyThing",
      "max": {
        "openshift.com/ShinyThingProperty": "5",
    },
    ],
  }
}
```

And enforcement of "openshift.com/ShinyThing" would happen in a separate LimitRange admission control plug-in for OpenShift, but from CLI, I would have a single resource to view my min/max constraints.